contracts/operators/Beefy/BeefyZapBiswapLPVaultOperator.sol

```diff
@@ -0,0 +1,287 @@
+        uint256 depositedAmount = tokenBalanceBefore - token.balanceOf(address(this));
+
+        require(vaultAmount != 0 && vaultAmount >= minVaultAmount, "BLVO: INVALID_AMOUNT_RECEIVED");
+        require(depositedAmount != 0 && amountToDeposit >= depositedAmount, "BLVO: INVALID_AMOUNT_DEPOSITED");
+
+        // Output amounts
+        amounts[0] = vaultAmount;
+        amounts[1] = depositedAmount;
+
+        // Output token
+        tokens[0] = vault;
+        tokens[1] = address(token);
+    }
+
+    /// @notice Withdraw the vault token (moo) from Beefy and receive
+    ///         one of the paired tokens
+    /// @param vault The vault address to withdraw from
+    /// @param amount The vault token amount to withdraw
+    /// @param token One of the paired token
+    /// @param minTokenAmount The minimum token amount expected
+    /// @return amounts Array of amounts :
+    ///         - [0] : The token received amount
+    ///         - [1] : The vault token deposited amount
+    /// @return tokens Array of token addresses
+    ///         - [0] : The token received address
+    ///         - [1] : The vault token deposited address
+    function withdraw(
+        address vault,
+        uint256 amount,
+        IERC20 token,
+        uint256 minTokenAmount
+    ) external returns (uint256[] memory amounts, address[] memory tokens) {
+        require(amount != 0, "BLVO: INVALID_AMOUNT");
+        address router = operatorStorage.vaults(vault);
+        require(router != address(0), "BLVO: INVALID_VAULT");
+
+        amounts = new uint256[](2);
+        tokens = new address[](2);
+
+        uint256 tokenBalanceBefore = token.balanceOf(address(this));
+        uint256 vaultBalanceBefore = IERC20(vault).balanceOf(address(this));
+
+        _withdrawAndSwap(router, vault, amount, address(token));
+
+        uint256 tokenAmount = token.balanceOf(address(this)) - tokenBalanceBefore;
+        uint256 vaultAmount = vaultBalanceBefore - IERC20(vault).balanceOf(address(this));
+        require(vaultAmount == amount, "BLVO: INVALID_AMOUNT_WITHDRAWED");
+        require(tokenAmount >= minTokenAmount, "BLVO: INVALID_OUTPUT_AMOUNT");
+
+        // Output amounts
+        amounts[0] = tokenAmount;
+        amounts[1] = amount;
+
+        // Output token
+        tokens[0] = address(token);
+        tokens[1] = vault;
+    }
+
+    /// @notice Perform a vault token withdraw (moo) from Beefy, and
+    ///         transfer the rest as one of the paired token.abi
+    /// @param router The uniswap v2 router address to use for swaping and adding liquidity
+    /// @param vault The vault address to withdraw from
+    /// @param amount The vault token amount to withdraw
+    /// @param token One of the paired token
+    function _withdrawAndSwap(
+        address router,
+        address vault,
+        uint256 amount,
+        address token
+    ) private {
+        IBiswapRouter02 biswapRouter = IBiswapRouter02(router);
```

Comment:
🔧 This can be declared further down in the scope to save gas fees if the execution reverts